spring-cloud-app-broker-acceptance-tests/src/test/java/org/springframework/cloud/appbroker/acceptance/CreateInstanceWithStackAcceptanceTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.acceptance;
+
+import java.util.Optional;
+
+import org.cloudfoundry.operations.applications.ApplicationDetail;
+import org.cloudfoundry.operations.applications.ApplicationSummary;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CreateInstanceWithStackAcceptanceTest extends CloudFoundryAcceptanceTest {
+
+	private static final String APP_CREATE_1 = "app-create-1";
+
+	private static final String SI_NAME = "si-create";
+
+	private static final String SUFFIX = "create-instance";
+
+	private static final String APP_SERVICE_NAME = "app-service-" + SUFFIX;
+
+	private static final String BACKING_SERVICE_NAME = "backing-service-" + SUFFIX;
+
+	@Override
+	protected String testSuffix() {
+		return SUFFIX;
+	}
+
+	@Override
+	protected String appServiceName() {
+		return APP_SERVICE_NAME;
+	}
+
+	@Override
+	protected String backingServiceName() {
+		return BACKING_SERVICE_NAME;
+	}
+
+	@Test
+	@AppBrokerTestProperties({
+		"spring.cloud.appbroker.services[0].service-name=" + APP_SERVICE_NAME,
+		"spring.cloud.appbroker.services[0].plan-name=" + PLAN_NAME,
+
+		"spring.cloud.appbroker.services[0].apps[0].name=" + APP_CREATE_1,
+		"spring.cloud.appbroker.services[0].apps[0].path=" + BACKING_APP_PATH,
+
+		"spring.cloud.appbroker.deployer.cloudfoundry.properties.stack=custom-stack"
+	})
+	void deployAppsOnCreateService() {
+		// when a service instance is created
+		createServiceInstance(SI_NAME);
+
+		// and stack is updated when specified
+		Optional<ApplicationDetail> application1Detail = getApplicationDetail(APP_CREATE_1);
```

Comment:
Not that the application won't start successfully since this stack does not exist, but we only care that the stack was changed.